File: Backend/src/main/java/com/example/PublicKeyInfrastructure/dto/certificate/CertificateDTO.java
package com.example.PublicKeyInfrastructure.dto.certificate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CertificateDTO {

    private String subjectCommonName;
    private String subjectOrganizationName;
    private String subjectOrganizationalUnit;
    private String subjectCountry;
    private String subjectEmail;
    private String serialNumber;
    private LocalDateTime validFrom;
    private LocalDateTime validTo;
}

File: Frontend/src/dto/certificate/CertificateDTO.ts
export interface CertificateDTO{
    subjectCommonName: string,
    subjectOrganizationName: string,
    subjectOrganizationalUnit: string,
    subjectCountry: string,
    subjectEmail: string,
    serialNumber: string, 
    validFrom: string,
    validTo: string,
    isDigitalSignature: boolean,
    isKeyEncipherment: boolean,
    isServerAuth: boolean,
    isClientAuth: boolean
}